pdns/syncres.cc

```diff
@@ -3418,9 +3424,16 @@ bool SyncRes::processRecords(const std::string& prefix, const DNSName& qname, co
         continue;
       }
     }
+    const bool negCacheIndiction = rec.d_place == DNSResourceRecord::AUTHORITY && rec.d_type == QType::SOA &&
+      lwr.d_rcode == RCode::NXDomain && qname.isPartOf(rec.d_name) && rec.d_name.isPartOf(auth);
+
+    bool putInNegCache = true;
+    if (negCacheIndiction && isForwardOrAuth(qname)) {
```

Comment:
Did you consider skipping the insertion only when the response comes from a DS query? For auth zones that probably doesn't matter but I'm wondering if we really want to disable this even when the response comes from the forwarded server?
Untested:
```suggestion
    if (negCacheIndiction && isForwardOrAuth(qname) && qtype == QType::DS) {
```

Reply:
Ugh, I had that in my head but neglected  to type that extra condition